### PR TITLE
Multihoming initial release

### PR DESCRIPTION
**Pull request by tymoteuszblochmobica**
### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [x] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### CODE REVIEW

**Comment by tymoteuszblochmobica** [comment]
@SeppoTakalo please review

**Comment by tymoteuszblochmobica** [comment]
@MikaLeppanen please review

**Comment by tymoteuszblochmobica** [comment]
@kjbracey-arm please review

**Comment by SeppoTakalo** [comment]
@mikaleppanen Can you review this please.

File: features/cellular/framework/API/CellularContext.h [hunk 1]
```diff
@@ -117,7 +117,7 @@ class CellularContext : public CellularBase {
      *  @param status_cb The callback for status changes.
      */
     virtual void attach(mbed::Callback<void(nsapi_event_t, intptr_t)> status_cb) = 0;
-    virtual nsapi_error_t connect() = 0;
+    virtual nsapi_error_t connect(bool is_default = false) = 0;
```
Comment (mikaleppanen) [inline features/cellular/framework/API/CellularContext.h]: See https://github.com/tymoteuszblochmobica/mbed-os/pull/7

File: features/lwipstack/LWIPInterface.cpp [hunk 2]
```diff
@@ -457,6 +474,30 @@ nsapi_error_t LWIP::add_l3ip_interface(L3IP &l3ip, bool default_if, OnboardNetwo
 nsapi_error_t LWIP::remove_l3ip_interface(OnboardNetworkStack::Interface **interface_out)
 {
 #if LWIP_L3IP
+    if((interface_out != NULL) && (*interface_out != NULL)){
+
+        Interface* lwip=  static_cast<Interface*>(*interface_out);
```
Comment (mikaleppanen) [inline features/lwipstack/LWIPInterface.cpp]: space after "lwip"

File: features/lwipstack/lwip/src/core/lwip_dns.c [hunk 3]
```diff
@@ -305,6 +305,7 @@ static struct dns_table_entry dns_table[DNS_TABLE_SIZE];
 static struct dns_req_entry   dns_requests[DNS_MAX_REQUESTS];
 #endif
 static ip_addr_t              dns_servers[DNS_MAX_SERVERS];
+struct dns_server_interface          *multihoming_dns_servers;
```
Comment (mikaleppanen) [inline features/lwipstack/lwip/src/core/lwip_dns.c]: indentation

File: features/lwipstack/lwip/src/core/lwip_dns.c [hunk 4]
```diff
@@ -366,14 +367,21 @@ dns_init(void)
...
+  if (netif == NULL || netif_check_default(netif)) {
+    if (numdns < DNS_MAX_SERVERS) {
+      if (dnsserver != NULL) {
+        dns_servers[numdns] = (*dnsserver);
+      } else {
+        dns_servers[numdns] = *IP_ADDR_ANY;
+      }
     }
+  } else {
+	    char name[INTERFACE_NAME_SIZE];
```
Comment (mikaleppanen) [inline features/lwipstack/lwip/src/core/lwip_dns.c]: indentation

File: features/lwipstack/lwip/src/include/lwip/dns.h [hunk 5]
```diff
@@ -103,8 +109,11 @@ typedef void (*dns_found_callback)(const char *name, const ip_addr_t *ipaddr, vo
 
 void             dns_init(void);
 void             dns_tmr(void);
-void             dns_setserver(u8_t numdns, const ip_addr_t *dnsserver);
-const ip_addr_t* dns_getserver(u8_t numdns);
+void             dns_setserver(u8_t numdns, const ip_addr_t *dnsserver, struct netif *netif);
+const ip_addr_t* dns_getserver(u8_t numdns, const char *interface_name);
+void dns_add_interface_server(u8_t numdns, const char *interface_name, const ip_addr_t *dnsserver);
```
Comment (mikaleppanen) [inline features/lwipstack/lwip/src/include/lwip/dns.h]: indentation

File: features/lwipstack/lwip/src/include/lwip/ip.h [hunk 6]
```diff
@@ -82,7 +82,9 @@ extern "C" {
    /* Type Of Service */ \
   u8_t tos;              \
   /* Time To Live */     \
-  u8_t ttl               \
+  u8_t ttl;               \
```
Comment (mikaleppanen) [inline features/lwipstack/lwip/src/include/lwip/ip.h:85]: extra space

File: features/lwipstack/LWIPInterface.cpp [hunk 7]
```diff
@@ -267,9 +272,21 @@ char *LWIP::Interface::get_mac_address(char *buf, nsapi_size_t buflen)
     return buf;
 }
 
-char *LWIP::Interface::get_ip_address(char *buf, nsapi_size_t buflen)
+char *LWIP::Interface::get_interface_name(char *buf)
+{
+    sprintf(buf, "%c%c%d", netif.name[0], netif.name[1], netif.num);
```
Comment (mikaleppanen) [inline features/lwipstack/LWIPInterface.cpp:277]: Make sure that netif structure has enough space for this (6 chars)

**Review by mikaleppanen — CHANGES_REQUESTED** [review]
Checked the pull request. Added some minor comments. Comments from this https://github.com/tymoteuszblochmobica/mbed-os/pull/7 needs to be addresses (DNS module asking IP address i.e. version from stack).

File: features/lwipstack/LWIPInterface.cpp (hunk 2 above)
Comment (tymoteuszblochmobica) [reply features/lwipstack/LWIPInterface.cpp]: done

File: features/lwipstack/lwip/src/core/lwip_dns.c (hunk 3 above)
Comment (tymoteuszblochmobica) [reply features/lwipstack/lwip/src/core/lwip_dns.c]: done

File: features/lwipstack/lwip/src/core/lwip_dns.c (hunk 4 above)
Comment (tymoteuszblochmobica) [reply features/lwipstack/lwip/src/core/lwip_dns.c]: done

File: features/lwipstack/lwip/src/include/lwip/dns.h (hunk 5 above)
Comment (tymoteuszblochmobica) [reply features/lwipstack/lwip/src/include/lwip/dns.h]: done

File: features/lwipstack/lwip/src/include/lwip/ip.h (hunk 6 above)
Comment (tymoteuszblochmobica) [reply features/lwipstack/lwip/src/include/lwip/ip.h:85]: done

File: features/lwipstack/LWIPInterface.cpp (hunk 7 above)
Comment (tymoteuszblochmobica) [reply features/lwipstack/LWIPInterface.cpp:277]: netif.num is u8_t

**Pull request closed without merging**